apps/events/api/tests/attendance_tests.py

```diff
@@ -0,0 +1,396 @@
+from django.contrib.auth.models import Group
+from django.utils import timezone
+from django_dynamic_fixture import G
+from onlineweb4.fields.recaptcha import mock_validate_recaptcha
+from rest_framework import status
+
+from apps.events.tests.utils import (
+    attend_user_to_event,
+    generate_event,
+    generate_payment,
+    generate_user,
+    pay_for_event,
+)
+from apps.online_oidc_provider.test import OIDCTestCase
+from apps.profiles.models import Privacy
+
+from ...models import Extras
+from .utils import generate_attendee
+
+
+class AttendanceEventTestCase(OIDCTestCase):
+    basename = "events_attendance_events"
+
+    def setUp(self):
+        self.committee = G(Group, name="Arrkom")
+        self.user = generate_user(username="_user")
+        self.privacy = G(Privacy, user=self.user)
+        self.token = self.generate_access_token(self.user)
+        self.headers = {
+            **self.generate_headers(),
+            "Accepts": "application/json",
+            "Content-Type": "application/json",
+            "content_type": "application/json",
+            "format": "json",
+        }
+        self.recaptcha_arg = {"recaptcha": "--mock-recaptcha--"}
+
+        self.event = generate_event(organizer=self.committee)
+        self.event.attendance_event.registration_start = timezone.now()
+        self.event.attendance_event.registration_end = timezone.now() + timezone.timedelta(
+            days=2
+        )
+        self.event.attendance_event.max_capacity = 20
+        self.event.attendance_event.save()
+        self.attendee1 = generate_attendee(self.event, "test1", "4321")
+        self.attendee2 = generate_attendee(self.event, "test2", "1234")
+        self.attendees = [self.attendee1, self.attendee2]
+
+    def get_register_url(self, event_id: int):
+        return self.get_action_url("register", event_id)
+
+    def get_unregister_url(self, event_id: int):
+        return self.get_action_url("unregister", event_id)
+
+    def get_attendee_url(self, event_id: int):
+        return self.get_action_url("attendee", event_id)
+
+    def get_public_attendees_url(self, event_id: int):
+        return self.get_action_url("public-attendees", event_id)
+
+    def get_extras_url(self, event_id: int):
+        return self.get_action_url("extras", event_id)
+
+    def get_payment_url(self, event_id: int):
+        return self.get_action_url("payment", event_id)
+
+    def test_anonymous_user_can_get_list(self):
+        response = self.client.get(self.get_list_url(), **self.bare_headers)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_anonymous_user_can_get_detail(self):
+        response = self.client.get(
+            self.get_detail_url(self.event.id), **self.bare_headers
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    @mock_validate_recaptcha()
+    def test_user_cannot_register_for_a_non_attendance_event(self, _):
```

Comment:
That argument comes from the mock decorator. It can be used to dynamically mock during the method call, so it should be possible to use it, so it has to be passed